SpatialGDK/Source/SpatialGDK/Public/LoadBalancing/LegacyLoadbalancingComponents.h

```diff
@@ -0,0 +1,185 @@
+#pragma once
+
+#include "Schema/StandardLibrary.h"
+#include "SpatialView/ComponentData.h"
+#include "SpatialView/ComponentUpdate.h"
+
+namespace SpatialGDK
+{
+struct LegacyLB_GridCell
+{
+	static constexpr Worker_ComponentId ComponentId = 190502;
+
+	LegacyLB_GridCell() {}
+
+	LegacyLB_GridCell(const ComponentData& Data)
+		: LegacyLB_GridCell(Data.GetUnderlying())
+	{
+	}
+
+	LegacyLB_GridCell(Schema_ComponentData* Data)
+	{
+		Schema_Object* ComponentObject = Schema_GetComponentDataFields(Data);
+		ReadFromObject(ComponentObject);
+	}
+
+	ComponentData CreateComponentData() const
+	{
+		ComponentData Data(ComponentId);
+		Schema_Object* ComponentObject = Schema_GetComponentDataFields(Data.GetUnderlying());
+
+		WriteToObject(ComponentObject);
+
+		return Data;
+	}
+
+	ComponentUpdate CreateComponentUpdate() const
+	{
+		ComponentUpdate Update(ComponentId);
+		Schema_Object* ComponentObject = Schema_GetComponentUpdateFields(Update.GetUnderlying());
+
+		WriteToObject(ComponentObject);
+
+		return Update;
+	}
+
+	void ApplyComponentUpdate(const ComponentUpdate& Update) { ApplyComponentUpdate(Update.GetUnderlying()); }
+
+	void ApplyComponentUpdate(Schema_ComponentUpdate* Update)
+	{
+		Schema_Object* ComponentObject = Schema_GetComponentUpdateFields(Update);
+		ReadFromObject(ComponentObject);
+	}
+
+	void ReadFromObject(Schema_Object* Object)
+	{
+		Center = Coordinates::ToFVector(GetCoordinateFromSchema(Object, 1));
+		Edge_length = Coordinates::ToFVector(GetCoordinateFromSchema(Object, 2));
+	}
+
+	void WriteToObject(Schema_Object* Object) const
+	{
+		AddCoordinateToSchema(Object, 1, Coordinates::FromFVector(Center));
+		AddCoordinateToSchema(Object, 2, Coordinates::FromFVector(Edge_length));
+	}
+
+	FVector Center;
+	FVector Edge_length;
+};
+
+struct LegacyLB_Layer
+{
+	static constexpr Worker_ComponentId ComponentId = 190501;
+
+	LegacyLB_Layer() {}
+
+	LegacyLB_Layer(const ComponentData& Data)
+		: LegacyLB_Layer(Data.GetUnderlying())
+	{
+	}
+
+	LegacyLB_Layer(Schema_ComponentData* Data)
```

Comment:
What would you say about making this a reference? We can at the very least guarantee it being non-null then.
```suggestion
	LegacyLB_Layer(Schema_ComponentData& Data)
```

Reply:
Maybe... a couple of templates are relying on it being a pointer, but I could change that. it is also inherited from the c-style APIs we have lingering around.